src/Jackett.Common/Indexers/SpeedCD.cs

```diff
@@ -111,79 +111,69 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
 
-            var qParams = new NameValueCollection();
-
-            if (!string.IsNullOrWhiteSpace(query.ImdbID))
-            {
-                qParams.Add("search", query.ImdbID);
-                qParams.Add("d", "on");
-            }
-            else if (!string.IsNullOrEmpty(query.GetQueryString()))
+            var qc = new List<KeyValuePair<string, string>>(); // NameValueCollection don't support c[]=30&c[]=52
+            if (query.IsImdbQuery)
             {
-                qParams.Add("search", query.GetQueryString());
+                qc.Add("search", query.ImdbID);
+                qc.Add("d", "on");
             }
+            else
+                qc.Add("search", query.GetQueryString());
 
             var catList = MapTorznabCapsToTrackers(query);
             foreach (var cat in catList)
-            {
-                qParams.Add("c" + cat, "1");
-            }
+                qc.Add("c[]", cat);
 
-            var urlSearch = SearchUrl;
-            if (qParams.Count > 0)
+            var searchUrl = SearchUrl + "?" + qc.GetQueryString();
+            var response = await RequestStringWithCookiesAndRetry(searchUrl);
+            if (!response.Content.Contains("/logout.php")) // re-login
             {
-                urlSearch += $"?{qParams.GetQueryString()}";
-            }
-
-            var response = await RequestStringWithCookiesAndRetry(urlSearch);
-            if (!response.Content.Contains("/logout.php"))
-            {
-                //Cookie appears to expire after a period of time or logging in to the site via browser
                 await DoLogin();
-                response = await RequestStringWithCookiesAndRetry(urlSearch);
+                response = await RequestStringWithCookiesAndRetry(searchUrl);
             }
 
             try
             {
                 var parser = new HtmlParser();
                 var dom = parser.ParseDocument(response.Content);
-                var rows = dom.QuerySelectorAll("div[id='torrentTable'] > div[class^='box torrentBox'] > div[class='boxContent'] > table > tbody > tr");
+                var rows = dom.QuerySelectorAll("div.boxContent > table > tbody > tr");
 
                 foreach (var row in rows)
                 {
                     var cells = row.QuerySelectorAll("td");
 
                     var title = row.QuerySelector("td[class='lft'] > div > a").TextContent.Trim();
-                    var link = new Uri(SiteLink + row.QuerySelector("img[title='Download']").ParentElement.GetAttribute("href").Trim());
-                    var comments = new Uri(SiteLink + row.QuerySelector("td[class='lft'] > div > a").GetAttribute("href").Trim().Remove(0, 1));
+                    var link = new Uri(SiteLink + row.QuerySelector("img[title='Download']").ParentElement.GetAttribute("href").Remove(0, 1));
+                    var comments = new Uri(SiteLink + row.QuerySelector("td[class='lft'] > div > a").GetAttribute("href").Remove(0, 1));
```

Comment:
In other trackers we use `.TrimStart('/')` instead of `Remove`

Reply:
changed